historydag/dag.py

```diff
@@ -271,6 +300,13 @@ def __eq__(self, other: object) -> bool:
         # identical trees return True. TODO
         raise NotImplementedError
 
+    def __getitem__(self, key) -> "HistoryDag":
+        r"""Returns the sub-history below the current history dag corresponding to the given index."""
+        self.count_trees()
+        if self.dagroot._dp_data < key:  # invalid index
+            return None
```

Comment:
I _think_ that the proper behavior when an index is out of bounds is to `raise IndexError` instead of returning `None`.

Proper handling of indices that are out of bounds is something we could test for.

We could also test that the syntax `for tree in hdag:` works to iterate through trees in the DAG, although this may require implementing more magic methods.